A3A/addons/core/functions/FactoryOverhaul/fn_productionGUIControls.sqf

```diff
@@ -12,6 +14,44 @@
 
 #define DEBUG 1
 
+FO_fn_GUI_itemCategories = [
+    "Any",
+    "Rhs Handguns",
+    "Rhs AssaultRifle",
+    "Rhs Launcher",
+    "Rhs Rifle",
+    "Rhs SniperRifle",
+    "Rhs SpecialWeapon",
+    "Rhs MachineGun",
+    "Rhs SubmachineGun",
+    "Rhs LauncherMagazine",
+    "Rhs Magazine",
+    "Rhs Optics",
+    "Rhs Muzzle",
+    "Rhs Underbarrel",
+    "Rhs Pointer",
+    "Rhs Navigation",
+    "Rhs Misc",
+    "Rhs Vest",
+    "Rhs Uniform",
+    "Rhs Backpack"
+];
+
+
+FO_fn_GUI_displayCategoryToItemCategory = {
+    params ["_displayCategory"];
+
+    private _itemCategory = "";
+    {
+        private _found = [_displayCategory, _x] call BIS_fnc_inString;
```

Comment:
Why you're using BIS_fnc_inString should probably have been explained in a comment, figuring out how this works was more effort than it should have been.

Reply:
Implementation is fine, though, it's just not very clear because one has to read through a lot more code should be necessary to finally figure out exactly what _displayCategory could be, and why we're searching *inside* strings.

But would this not cause problems with non-rhs weapons?
A vanilla weapon's category might be "Rifle", which is going to return true on 
```sqf
["Rifle", "Rhs Rifle"] call BIS_fnc_inString;
```

If this is not the case, let me know why.

Reply:
Honest to god I should just abolish the "Rhs" prefix, so it finds every weapon, vanilla and non, of that particular category.

Reply:
One more note, for the current implementation, checking if the strings are equal is enough, don't need to use BIS_fnc_inString. However, I'm gonna keep it there for potential further application.